### PR TITLE
fix: only build xtask once for the tests

### PR DESCRIPTION
**Pull request by Natoandro**
Use the xtask binary to run the tests.

- [ ] The change comes with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

## Summary by CodeRabbit

- **New Features**
  - Updated platform compatibility to `x86_64-linux`.
  - Added new configuration entry for enhanced versioning and platform support.

- **Improvements**
  - Modified test script to use a custom build script for better test management.

These changes improve platform compatibility and streamline the testing process.

### CODE REVIEW

**Review by afmika — APPROVED** [review]

**Comment by Yohe-Am** [comment]
Do the same for meta-cli as well? Each test file "builds" it once with the current scheme.

**Comment by Yohe-Am** [comment]
Also, eliminate the `deno task test` call and go to `cargo x deno test` directly.

**Review by zifeo — APPROVED** [review]
Nice improvement! @Yohe-Am's suggestion is great also, wanna tackle here or next?

**Comment by Natoandro** [comment]
> Nice improvement! @Yohe-Am's suggestion is great also, wanna tackle here or next?



> Nice improvement! @Yohe-Am's suggestion is great also, wanna tackle here or next?

I'll add it here...

File: dev/test.ts [hunk 1]
```diff
@@ -106,6 +106,7 @@ const env: Record<string, string> = {
   "TMP_DIR": tmpDir,
   "TIMER_MAX_TIMEOUT_MS": "30000",
   "NPM_CONFIG_REGISTRY": "http://localhost:4873",
+  "PATH": `${Deno.env.get("PATH")}:${join(projectDir, "target/debug")}`,
```
Comment (Yohe-Am) [inline dev/test.ts:109]: You probably ought to prefix the build dir path since we want items in there to be resolved first.

**Review by Yohe-Am — APPROVED** [review]

**Review by destifo — APPROVED** [review]

**Pull request merged**